modules/navier_stokes/test/tests/finite_volume/ins/channel-flow/segregated/2d/2d-segregated-scalar.i

```diff
@@ -0,0 +1,280 @@
+    two_term_boundary_expansion = false
+  []
+  [vel_y]
+    type = INSFVVelocityVariable
+    initial_condition = 0.0
+    nl_sys = v_system
+    two_term_boundary_expansion = false
+  []
+  [pressure]
+    type = INSFVPressureVariable
+    nl_sys = pressure_system
+    initial_condition = 0.2
+    two_term_boundary_expansion = false
+  []
+  [scalar_1]
+    type = INSFVScalarFieldVariable
+    nl_sys = scalar_1_system
+    initial_condition = 1.2
+  []
+  [scalar_2]
+    type = INSFVScalarFieldVariable
+    nl_sys = scalar_2_system
+    initial_condition = 1.2
+  []
+[]
+
+[FVKernels]
+  [u_advection]
+    type = INSFVMomentumAdvection
+    variable = vel_x
+    advected_interp_method = ${advected_interp_method}
+    velocity_interp_method = ${velocity_interp_method}
+    rho = ${rho}
+    momentum_component = 'x'
+  []
+  [u_viscosity]
+    type = INSFVMomentumDiffusion
+    variable = vel_x
+    mu = ${mu}
+    momentum_component = 'x'
+  []
+  [u_pressure]
+    type = INSFVMomentumPressure
+    variable = vel_x
+    momentum_component = 'x'
+    pressure = pressure
+    extra_vector_tags = ${pressure_tag}
+  []
+  [v_advection]
+    type = INSFVMomentumAdvection
+    variable = vel_y
+    advected_interp_method = ${advected_interp_method}
+    velocity_interp_method = ${velocity_interp_method}
+    rho = ${rho}
+    momentum_component = 'y'
+  []
+  [v_viscosity]
+    type = INSFVMomentumDiffusion
+    variable = vel_y
+    mu = ${mu}
+    momentum_component = 'y'
+  []
+  [v_pressure]
+    type = INSFVMomentumPressure
+    variable = vel_y
+    momentum_component = 'y'
+    pressure = pressure
+    extra_vector_tags = ${pressure_tag}
+  []
+  [p_diffusion]
+    type = FVAnisotropicDiffusion
+    variable = pressure
+    coeff = "Ainv"
+    coeff_interp_method = 'average'
+  []
+  [p_source]
+    type = FVDivergence
+    variable = pressure
+    vector_field = "HbyA"
+    force_boundary_execution = true
+  []
+
+  [scalar_1_advection]
+    type = INSFVScalarFieldAdvection
+    variable = scalar_1
+    velocity_interp_method = ${velocity_interp_method}
+    advected_interp_method = ${advected_interp_method}
+  []
+  [scalar_1_diffusion]
+    type = FVDiffusion
+    coeff = ${diff}
+    variable = scalar_1
+  []
+  [scalar_1_src]
+    type = FVBodyForce
+    variable = scalar_1
+    value = 1.0
+  []
+  [scalar_1_coupled_source]
+    type = FVCoupledForce
+    variable = scalar_1
+    v = scalar_2
+    coef = 0.1
+  []
+
+  [scalar_2_advection]
+    type = INSFVScalarFieldAdvection
+    variable = scalar_2
+    velocity_interp_method = ${velocity_interp_method}
+    advected_interp_method = ${advected_interp_method}
+  []
+  [scalar_2_diffusion]
+    type = FVDiffusion
+    coeff = '${fparse 2*diff}'
+    variable = scalar_2
+  []
+  [scalar_2_src]
+    type = FVBodyForce
+    variable = scalar_2
+    value = 5.0
+  []
+  [scalar_2_coupled_source]
+    type = FVCoupledForce
+    variable = scalar_2
+    v = scalar_1
+    coef = 0.05
+  []
```

Comment:
when turning those terms off I still dont get conservation for the scalars until I turn off diffusion
do you know why?

Reply:
Are you running a symmetry-based test? For those we need to enforce the boundary execution on the symmetry plane. I'll push the changes soon!

Reply:
For the advection term we determine the face velocities using the segregated RC interpolation. Even though it has a symmetry BC (which would indicate that there is no cross flow) we still see some residual cross flow when we use the segregated RC to determine the face velocity. That needs to be accounted for in the advection terms (so we need to enforce the execution on symmetry boundaries).

Reply:
I'll take a look at the conservation here.

Reply:
I disable the source terms and get perfect conservation based on the postprocessors. Added a test for this too.